docs/rfcs/040-neon-supported-extensions.md

```diff
@@ -0,0 +1,140 @@
+## Goals:
+
+- Support multiple versions of extensions available in compute image
+    - This is needed to support installed extensions that are not backward compatible.
+    - This will be useful for [multi-version compute images](https://github.com/neondatabase/neon/issues/6685).
+    - This will be useful to test new versions of extensions before updating them in the compute image.
+
+- Allow overriding extension version for a specific project/user.
+
+- Uncouple extension version from compute build_tag to be able to release them with different pace
+
+- Support private extensions.
+
+- Improve visibility of extension versions used in our compute images.
+
+- Automate extension updates as much as possible: 
+    - track available versions
+    - run backward compatibility tests
+    - prepare update PR (if tests pass)
+
+## Out of scope goals:
+
+- Ensure compute image rollbacks are safe.
+
+    Imagine, there is extension that adds function `foo_2()` in release 1.2. If user creates extension with version 1.2 and after that we rollback to compute image that only contains extension 1.1, `foo_2()` call will fail in a newly started compute. We only rollback compute if release went wrong, and rollback is supposed to be short-living, so this is not an urgent isue.
+    Though, new design, allows manual cplane table editing to map extension version to compute version.
+
+- Maintain diffferent system library versions that are needed for different extension versions. I.e. `my_ext-1.0` depends on `libicu42` and `my_ext-2.0` depends on `libicu52` (numbers are random).
+    TODO: find at least one example of such issue.
+
+## Involved components:
+
+- cplane
+- compute
+- postgres
+- build compute image CI job
+- build-custom-extensions CI job
+- remote extensions s3 bucket
+- extension testing infrastructure
+- backward compatibility testing infrastructure
+
+
+## Existing architecture
+
+We have 2 ways to support extensions:
+1. Put extensions directly into [Dockerfile.compute-node](https://github.com/neondatabase/neon/blob/main/Dockerfile.compute-node)
+2. use  [custom extensions](https://github.com/neondatabase/build-custom-extensions/blob/main/README.md)
+
+## Proposed solution:
+
+TLDR: Improve metadata handling and testing for custom extensions and move all extensions to this workflow.
+
+### 1. Keep mapping of supported extension versions in a cplane table:
+
+Create new cplane table:
+
```

Comment:
TODO: rewrite this to avoid using cplane.

Now, we can keep with extension-compute 1-1  version mapping and put supported_extension manifest, generated in `build-custom-extensions` CI into s3 and/or into compute manifest / and/or directly into compute image.